### PR TITLE
 droidcam: initial inclusion at 2.1.2

### LINKED ISSUE #282: DroidCam (T10161)
<div dir="rtl"><b>Leo</b>&nbsp;(<em>&#35;528491</em>),&nbsp;2022-02-08&nbsp;14:42:16&nbsp;UTC</div><br/>

**Name:**
DroidCam

**Homepage:**
https://www.dev47apps.com/

**Why should this be included in the repository?**
There is scrcpy, but DroidCam adds:
- Usability (GUI)
- Audio support (use phone as webcam with microphone)

**Is it Open Source**
yes

**Who and how many users do you anticipate will use this software?**
People who have an android phone but no (or a bad) webcam and want to use a webcam + microphone on Solus

**Link to source tarball/zip file**
https://github.com/dev47apps/droidcam/releases

There has been a request some time ago, but back then there were no versioned downloads. Now there are .zip versioned donwloads on GitHub.

### PR DESCRIPTION
**Pull request by malfisya**
**Summary**

Initial inclusion of droidcam at 2.1.2
- Resolves #282 

**Test Plan**
- Android
    - [x] Connect via wifi 
    - [x] Connect via usb
    - [x] Video only
    - [x] Audio only
    - [x] Video and audio
- IOS 
  - [ ] Connect via wifi 
  - [ ] Connect via usb
  - [ ] Video only
  - [ ] Audio only
  - [ ] Video and audio
- Test with video conferencing app e.g : zoom, teams

**Checklist**

- [x] Package was built and tested against unstable

**Comment**
- Needed someone with IOS device to test this. It still complain about `usbmuxd` service not started
- ~~Still couldn't figure out to make this split to `droidcam` and `droidcam-current` so can install different `v4l2loopback` version~~  ~~made two recipes~~
Back to one recipe with link to instructions in the description.

### CODE REVIEW

**Comment by malfisya** [comment]
New consideration : should i just make this to only one repice but with both of `v4l2loopback` and `v4l2loppback-current` as rundeps. As it appears both version co-installable. This mean users can just switch kernel without uninstalling and installing different package each time.

**Comment by malfisya** [comment]
Now only one recipe

**Comment by malfisya** [comment]
Still need review for this one

File: packages/d/droidcam/files/droidcam.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=DroidCam
+Comment=Use your phone as a webcam
+TryExec=droidcam
+Exec=/usr/bin/droidcam
+Icon=droidcam
+Terminal=false
+Type=Application
+Categories=Video;AudioVideo;
```
Comment (ReillyBrogan) [inline packages/d/droidcam/files/droidcam.desktop:9]: Please add a trailing newline.

File: packages/d/droidcam/files/modprobe.conf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+options v4l2loopback video_nr=0
+options v4l2loopback card_label="droidcam"
+options v4l2loopback exclusive_caps=1
+options v4l2loopback max_width=1920
+options v4l2loopback max_height=1080
+options snd-aloop index=0
```
Comment (ReillyBrogan) [inline packages/d/droidcam/files/modprobe.conf:6]: Add trailing newline.

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]
I was skimming the Github repo for droidcam and noticed [this issue](https://github.com/dev47apps/droidcam/issues/262). Given that we're on kernel 6.6 you should probably pull that fix.

**Comment by malfisya** [comment]
Version 2.0.0 rebuild and tested on kernel 6.6.7.  Run with this [behaviour](https://github.com/dev47apps/droidcam/issues/242) 
Version 2.1.1 build and tested on kernel 6.6.7. Didn't launch with this error :
`Fatal: Unable to query video device. dev=/dev/video0 errno=22`.

There is also [droidcam-obs](https://github.com/dev47apps/obs-studio), which probably overkill.

**Comment by malfisya** [comment]
Now at 2.1.2 with previous issues resolved.

File: packages/d/droidcam/package.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+source     :
+    - https://github.com/dev47apps/droidcam/archive/refs/tags/v2.1.2.tar.gz : db65ccabc2f54bcd581963e2c36c56a611eea45e5332a39d24440b0966a704b4
+homepage   : http://www.dev47apps.com/
+license    : GPL-2.0-or-later
+component  : multimedia.video
+summary    : GNU/Linux client for DroidCam
+description: |
+    DroidCam turns your phone/tablet into a webcam for your PC. Use it with chat programs like Zoom, MS Teams, and Skype.
+
+    Installation Guide : Please install `v4l2loopback-current` if you are on default kernel(`linux-current`) or `v4l2loopback` if you are on LTS kernel(`linux-lts`).
```
Comment (ReillyBrogan) [inline packages/d/droidcam/package.yml:13]: We don't put instructions in the description, this would be better in the help center.

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]
My only remaining nit, excellent job!

File: packages/d/droidcam/package.yml (hunk 3 above)
Comment (malfisya) [reply packages/d/droidcam/package.yml:13]: How do i go about this? Remove instructions for now? Add instruction later when it is added in the help center? I still need to learn about making PR in help center.

Comment (ReillyBrogan) [reply packages/d/droidcam/package.yml:13]: Probably drop the instructions for now, I can merge without them. Perhaps add something in [this folder](https://github.com/getsolus/help-center-docs/tree/master/docs/user/hardware)?

**Comment by malfisya** [comment]
Docs PR https://github.com/getsolus/help-center-docs/pull/444

**Comment by malfisya** [comment]
Docs added to description. Ready to merge.

**Review by ReillyBrogan — APPROVED** [review]

**Pull request merged**